lib/src/log.rs

```diff
@@ -129,22 +130,84 @@ pub fn reset_task_log_directory(path: &Path) -> Result<(), Error> {
     Ok(())
 }
 
+pub fn read_last_lines_as_byte_deque(file: &mut File, lines: usize) -> Result<VecDeque<u8>, Error> {
+    // The integer types were chosen to avoid overflows when converting between signed and unsigned
+    // integers. u16 is used here so that we can support 16bit platforms with simple integer casts
+    // (FIXME Not sure if pueue even is supported on any 16bit platforms)
+    let chunk_size = 4096u16;
+    let mut buf = vec![0u8; chunk_size as usize];
+    let size = match file.seek(io::SeekFrom::End(0)) {
+        Ok(p) => p,
+        Err(err) => {
+            return Err(Error::LogSeek(format!(
+                "Error while seeking log file : {}",
+                err
+            )));
+        }
+    };
+    let mut position = size;
+    let mut text = std::collections::VecDeque::new();
+    let mut lines_seen = 0usize;
+    'outer: loop {
+        if position == 0 {
+            break;
+        }
+        // Safe to cast chunk_size from u16 to u64
+        let next_chunk_size = if position > chunk_size as u64 {
+            chunk_size
+        } else {
+            // Safe to cast position to u16 since it is <= chunk_size which is u16
+            position as u16
+        };
+        // Safe to cast next_chunk_size from u16 to i64
+        position = match file.seek(io::SeekFrom::Current(-(next_chunk_size as i64))) {
```

Comment:
This is very equal to the current behavior of https://doc.rust-lang.org/std/io/struct.BufReader.html.

I checked whether there already is a crate for this and it turns out there is!
https://docs.rs/rev_buf_reader/latest/rev_buf_reader/

The project is rather small, but it is very well documented and most importantly properly tested.

To be honest, I don't feel too comfortable to include such a finnicky algorithm, if there's a proper and tested implementation out there. Just like the official BufReader it can also read lines in reverse.

I made a small playground project:

Cargo.toml:
```
[package]
name = "rofl"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
rev_buf_reader = "0.2"
```

main.rs: 
```
use rev_buf_reader;

use rev_buf_reader::RevBufReader;
use std::io::{self, BufRead};

fn main() {
    let data = "This\nis a long\ntest sentence";
    let inner = io::Cursor::new(&data);
    let reader = RevBufReader::new(inner);
    let mut lines = reader.lines();

    assert_eq!(lines.next().unwrap().unwrap(), "test sentence".to_string());
    assert_eq!(lines.next().unwrap().unwrap(), "is a long".to_string());
    assert_eq!(lines.next().unwrap().unwrap(), "This".to_string());
    assert!(lines.next().is_none());
}
```

Reply:
This implementation will be a little less performant as we're no longer operating on the byte level, but also introduce UTF-8 handling. However, performance usually isn't an issue in these cases. A few `μs` on top won't hurt us.

We should also be able to safely use the `utf8_from_lossy` function. It replaces invalid characters with the Placeholder character, which would be done by the terminal lateron anyway.